cpp/examples/interop/interop.cpp

```diff
@@ -0,0 +1,159 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "arrow/array/array_binary.h"
+#include "arrow/type.h"
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/interop.hpp>
+
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+// Helper functuons to create StringViews
+inline arrow::BinaryViewType::c_type ToInlineStringView(const void* data, int32_t size)
+{
+  arrow::BinaryViewType::c_type out;
+  out.inlined = {size, {}};
+  memcpy(&out.inlined.data, data, size);
+  return out;
+}
+inline arrow::BinaryViewType::c_type ToInlineStringView(std::string_view v)
+{
+  return ToInlineStringView(v.data(), static_cast<int32_t>(v.size()));
+}
+inline arrow::BinaryViewType::c_type ToStringView(const void* data,
+                                                  int32_t size,
+                                                  int32_t buffer_index,
+                                                  int32_t offset)
+{
+  if (size <= arrow::BinaryViewType::kInlineSize) { return ToInlineStringView(data, size); }
+  arrow::BinaryViewType::c_type out;
+  out.ref = {size, {}, buffer_index, offset};
+  memcpy(&out.ref.prefix, data, sizeof(out.ref.prefix));
+  return out;
+}
+inline arrow::BinaryViewType::c_type ToStringView(std::string_view v,
+                                                  int32_t buffer_index,
+                                                  int32_t offset)
+{
+  return ToStringView(v.data(), static_cast<int32_t>(v.size()), buffer_index, offset);
+}
+
+/**
+ * @brief Create a StringViewArray
+ *
+ * @param data_buffers The data buffers
+ * @param views The string views
+ * @param validate Whether to validate the array
+ */
+arrow::Result<std::shared_ptr<arrow::StringViewArray>> MakeStringViewArray(
+  arrow::BufferVector data_buffers,
+  const std::vector<arrow::BinaryViewType::c_type>& views,
+  bool validate = true)
+{
+  auto length = static_cast<int64_t>(views.size());
+  auto arr    = std::make_shared<arrow::StringViewArray>(
+    arrow::utf8_view(), length, arrow::Buffer::FromVector(views), std::move(data_buffers));
+  if (validate) { RETURN_NOT_OK(arr->ValidateFull()); }
+  return arr;
+}
+
+/**
+ * @brief Convert a vector of strings into a vector of the
+ * constituent chars and a vector of offsets of the strings
+ * in the chars vector
+ *
+ * @param strings The vector of strings
+ */
+auto make_chars_and_offsets(std::vector<std::string> strings)
+{
+  std::vector<char> chars{};
+  std::vector<cudf::size_type> offsets(1, 0);
+  for (auto& str : strings) {
+    chars.insert(chars.end(), std::cbegin(str), std::cend(str));
+    auto const last_offset = static_cast<std::size_t>(offsets.back());
+    auto const next_offset = last_offset + str.length();
+    CUDF_EXPECTS(
+      next_offset < static_cast<std::size_t>(std::numeric_limits<cudf::size_type>::max()),
+      "Cannot use strings_column_wrapper to build a large strings column");
```

Comment:
Looks like you copied this from the `strings_column_wrapper` code. You'll need a different exception message.

Reply:
I think we can remove the error message here

Reply:
Nevermind, let's keep it.